### DISCUSSION TITLE
Cross platform desktop application

### DISCUSSION THREAD

**Discussion by RutulPatel8** [dotnet/aspnetcore#58131] [General]
How to create a cross platform desktop application? Which works on linux, mac, and windows platforms?

I tried maui already but it only supports mac-catalysts not mac.

Can we do these with a .net core or do we need to choose another framework like electron or something..

**Comment by martincostello** [comment]
This isn't a question related to ASP.NET Core.

**Reply by RutulPatel8** [reply]
Yes are these for .net only. If dotnet isn't compatible with that. We need to look for another solution

**Comment by stephenjeffreywhite** [comment]
@RutulPatel8  Take a look at Photino if you want to use Blazor / ASP.NET.

https://www.tryphotino.io/

https://docs.tryphotino.io/Photino-Blazor

Hope this helps.

**Comment by SommerEngineering** [comment]
I did this for [AI Studio](https://github.com/MindWorkAI/AI-Studio): The UI is implemented using ASP.NET (Blazor) using the MudBlazor library. Since I also wanted to support the Linux desktop, and Microsoft currently does not support this (see e.g. https://github.com/dotnet/maui/issues/11738 and https://github.com/dotnet/maui/issues/5801), I chose Rust together with [Tauri](https://tauri.app/). So, there is a thin runtime layer written in Rust. Tauri then creates and manages the window. Works wonderfully 👍.

**Comment by WeihanLi** [comment]
Maybe could have a try on https://avaloniaui.net/

https://github.com/avaloniaui/avalonia
